src/cocotb/decorators.py

```diff
@@ -166,7 +168,7 @@ def test(
     expect_error: Union[bool, Type[Exception], Sequence[Type[Exception]]] = (),
     skip: bool = False,
     stage: int = 0,
-) -> Callable[[Callable[..., Coroutine[Any, Any, None]]], Test]:
+) -> Callable[[Callable[..., Coroutine[Any, Any, None]]], Test | None]:
```

Comment:
Are we allowing the | operator on types? I believe this raises the minimum python version to 3.8 or something like that

Reply:
This becomes possible with `from __future__ import annotations` which is supported since Python 3.7 (https://docs.python.org/3/library/__future__.html).

Reply:
I'm going to use optional here for 3.6